plugin/src/main/java/org/opensearch/ml/cluster/MLSyncUpCron.java

```diff
@@ -196,6 +201,24 @@ void refreshModelState(Map<String, Set<String>> modelWorkerNodes, Map<String, Se
                     int currentWorkerNodeCountInIndex = sourceAsMap.containsKey(MLModel.CURRENT_WORKER_NODE_COUNT_FIELD)
                         ? (int) sourceAsMap.get(MLModel.CURRENT_WORKER_NODE_COUNT_FIELD)
                         : 0;
+                    boolean deployToAllNodes = sourceAsMap.containsKey(MLModel.DEPLOY_TO_ALL_NODES_FIELD)
+                        ? (boolean) sourceAsMap.get(MLModel.DEPLOY_TO_ALL_NODES_FIELD)
+                        : false;
+                    List<String> planningWorkNodes = sourceAsMap.containsKey(MLModel.PLANNING_WORKER_NODES_FIELD)
+                        ? (List<String>) sourceAsMap.get(MLModel.PLANNING_WORKER_NODES_FIELD)
+                        : new ArrayList<>();
+                    if (deployToAllNodes) {
+                        DiscoveryNode[] eligibleNodes = nodeHelper.getEligibleNodes();
+                        planningWorkerNodeCount = eligibleNodes.length;
+                        List<String> eligibleNodeIds = Arrays
+                            .asList(eligibleNodes)
+                            .stream()
+                            .map(n -> n.getId())
+                            .collect(Collectors.toList());
+                        if (eligibleNodeIds.size() != planningWorkNodes.size() || !eligibleNodeIds.containsAll(planningWorkNodes)) {
```

Comment:
For the first condition, as an example, eligibleNodeIds size is 3 with nodeIDs node1, node2, node3. And planningWorkNodes size is 2 with nodeIDs node1, node2. 
Should we update newPlanningWorkerNodes with eligibleNodeIds to add node3 on planningNodes?